src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> sessionQuery = session
+                    .createQuery("from MovieSession m "
+                            + "join fetch m.movie mm where m.id = :id ", MovieSession.class);
```

Comment:
It is better to write all SQL commands in upper case, fix everywhere
```suggestion
                    .createQuery("FROM MovieSession m "
                            + "JOIN FETCH m.movie mm WHERE m.id = :id ", MovieSession.class);
```